v2ray/vmess/listener.go

```diff
@@ -0,0 +1,67 @@
+package vmess
+
+import (
+	"context"
+	"net"
+
+	vmess "github.com/getlantern/sing-vmess"
+	N "github.com/getlantern/sing-vmess/network"
+	"github.com/sagernet/sing/common/metadata"
+)
+
+type listener struct {
+	net.Listener
+	service *vmess.Service[int]
+}
+
+func NewVMessListener(baseListener net.Listener, uuids []string) (net.Listener, error) {
```

Comment:
Can you add documentation?  Specifically, what are the UUIDs?  How are we supposed to know what to configure here?

Reply:
Looking at the test, it appears that each client would have to present its own UUID which needs to be pre-configured on the proxy.  Is that correct?  How do you plan to achieve that?

Reply:
@hwh33 each proxy is configured with X UUIDs.
A random UUID from those X is returned when client requests proxy config.

I'll add a comment

Reply:
What happens when two clients are assigned the same UUID for a proxy?  Or do we just assign so many that the odds of that are negligible?

Reply:
@hwh33 no, it's no 1-to-1, think of it more of a 'password' and less 'user-id'.
you can have just one, but I think adding some entropy is always nicer

Reply:
There's still no documentation @reflog ☹️

Reply:
@hwh33 doh. slipped my mind. will add in a followup PR

Reply:
Thank you!